### DISCUSSION TITLE
HIP 367 - Removing Token association limit per Account.

### DISCUSSION THREAD

**Discussion by anighanta** [hiero-ledger/hiero-improvement-proposals#380] [Edits]
Discussion for #367

**Comment by hashaxis** [comment]
In favor of removing this limit or increasing it.
Would like to echo our comments from here: https://github.com/hashgraph/hedera-improvement-proposal/discussions/107
--

1,000 max associations is far too low for even our most basic use cases for a treasury account. Even a consumer grade user could easily reach 1,000 tokenIds associated when mostly dealing in "1-of-1" NFTs (such as collectibles or other unique items used to verify memberships or ownerships.)

We understand the concern about flooding the network with cheap airdrops; and propose a scaling fee model where the fee increase on some scale as you enter new tiers of associated accounts.

Scaling Fee Example:
0 - 1000 (standard fees)
1001 - 10,000 (standard fees) * (1.10)
10,001 - 50,000 (standard fees) * (1.50)
50,001 - 100,000 (standard fees) * (2)

A scaling fee model will allow enterprise cases who are willing to pay higher fees to do so, and discourage bad actors who will find it infinitely more expensive the more they try to flood the network.

Then Enterprise customers can make a business decision to calculate if their product fees cover the cost of associating more accounts. (Currently the only option we have is to start juggling potentially thousands of new accounts unnecessarily which from a security and development standpoint is a waste.)

**Reply by rbair23** [reply]
Hi, thanks for the detailed comment. I think we have two issues:
1. Handling auto-association
2. Association limit

This HIP is squarely focused on #2 "Association limit".  And it does this by getting rid of the limit entirely and without changing the fee schedule (the council of course could adopt the very reasonable fee amplification concept you had here, but this HIP doesn't plan to introduce it).

**Comment by topshef** [comment]
i don't understand this hip, who needs it? the motivation statement is missing. a token ID can already hold unlimited NFTs 

If you need more than 1000 collections then just open a new account it's cheap and more secure to distribute the assets anyway
